### ISSUE TITLE
installing as root does not fail (or: prints an over-severe warning) 

### ISSUE THREAD

**Issue by jamesbroadhead** [NixOS/nix#1125]
If installed as root, the following warning is printed: 
"warning: installing Nix as root is not supported by this script!" 
The install then continues. 

As it stands, the warning worries the user without explaining the consequences or likely outcomes. 

This warning is either too severe (root installs are supported) or it should be promoted to an error & the install should fail at this check. 

Supporting comment: https://github.com/NixOS/nix/issues/697#issuecomment-228298408

Alternately, support root installs: #936

**Comment by edolstra** [comment]
It's a warning because some people *do* use this script to do root installs, but you have to know what you're doing.

**Comment by jamesbroadhead** [comment]
Yep, I understand -- but the wording implies that it won't work. 

Perhaps something like: 
"warning: although it has some users, installing Nix as root is not supported by this script. Please see <link> for details. This install will continue, but we recommend aborting and installing as a user instead."

**Comment by copumpkin** [comment]
Maybe file a PR with the wording clarified?

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
